SMP-induced build failure for  GHC 8.4.1

```
  [ 3 of 30] Compiling Codec.Picture.Metadata ( src/Codec/Picture/Metadata.hs, dist/build/Codec/Picture/Metadata.o )

  src/Codec/Picture/Metadata.hs:172:10: error:
      • No instance for (Semigroup Metadatas)
          arising from the superclasses of an instance declaration
      • In the instance declaration for ‘Monoid Metadatas’
      |
  172 | instance Monoid Metadatas where
      |          ^^^^^^^^^^^^^^^^
  builder for ‘/nix/store/gqjih589yd4cpizisldlb66brkx4k643-JuicyPixels-3.2.9.3.drv’ failed with exit code 1
```

--- a/src/Codec/Picture/Metadata.hs
+++ b/src/Codec/Picture/Metadata.hs
@@ -171,7 +171,12 @@ newtype Metadatas = Metadatas
 
 instance Monoid Metadatas where
   mempty = empty
+#if !MIN_VERSION_base(4,11,0)
   mappend = union
+#else
+instance Semigroup Metadatas where
+  (<>) = union
+#endif
 
 -- | Right based union
 union :: Metadatas -> Metadatas -> Metadatas

